app/client/lib/textUtils.ts

```diff
@@ -32,18 +32,56 @@ For 'Link (in http://www.uk?)'
 'url-regex' [ 'http://www.uk?)' ]
 */
 
-// Match http or https then domain name (with optional port) then any text that ends with letter or number.
-export const urlRegex = /(https?:\/\/[A-Za-z\d][A-Za-z\d-.]*(?!\.)(?::\d+)?(?:\/[^\s]*)?[\w\d/])/;
+// Match http or https then domain name and capture markdown link text and URL separately
+const urlRegex = /(?:\[(.*?)\]\()?https?:\/\/[A-Za-z\d][A-Za-z\d-.]*\.[A-Za-z]{2,}(?::\d+)?(?:\/[^\s)]*)?(?:\))?/;
```

Comment:
I think this would match strings like `[Google](https://www.google.com` or `https://www.google.com)`.

Reply:
You are right @georgegevoian - the regex needs fix;
I will take a look how can I add tests. 

Btw one of the CI tests of nbrowser is failing and I can't understand why.

Reply:
`DetailView does not opens cell for editing when clicked on link` looks to be failing because a localhost URL isn't being treated as a link (most likely a regression in the regex).

The other failures look unrelated - we'll be landing a fix on `main` soon.

Thanks.